systems/admin/admin-adapter-out/src/main/kotlin/hs/kr/entrydsm/admin/adapterout/storage/S3StorageAdapter.kt

```
@@ -0,0 +1,63 @@
+            ).url().toExternalForm()
+        }.getOrElse { cause ->
+            throw AdminDomainException(ErrorCode.STORAGE_UNAVAILABLE, cause)
+        }
+
+    override fun exists(objectKey: String): Boolean =
+        try {
+            s3Client.headObject(
+                HeadObjectRequest.builder().bucket(bucket).key(objectKey).build(),
+            )
+            true
+        } catch (exception: NoSuchKeyException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)